### PR TITLE
Add additional bukkit dependency error

### PR DESCRIPTION
**Pull request by JulianVennen**

### CODE REVIEW

File: lang/en.json [hunk 1]
```diff
@@ -12,6 +12,7 @@
   "pex-config-problem": "The configuration of the plugin 'PermissionsEx' is invalid.",
   "ambiguous-plugin-name-problem": "There are multiple plugin files for the plugin name '{{plugin-name}}': '{{first-plugin-path}}' and '{{second-plugin-path}}'.",
   "plugin-dependency-problem": "The plugin '{{plugin-name}}' is missing the required the plugin '{{dependency-plugin-name}}'.",
+  "plugin-dependencies-problem": "The plugin '{{plugin-name}}' is missing the required the plugins '{{dependency-plugin-names}}'.",
```
Comment (matthi4s) [inline lang/en.json:15]:
```suggestion
  "plugin-dependencies-problem": "The plugin '{{plugin-name}}' is missing the required plugins '{{dependency-plugin-names}}'.",
```

File: src/Analysis/Problem/Bukkit/PluginDependenciesProblem.php [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+     * Apply the matches from the pattern
+     *
+     * @param array $matches
+     * @param $patternKey
+     */
+    public function setMatches(array $matches, $patternKey): void
+    {
+        $this->pluginPath = $matches[1];
+        $this->pluginName = $matches[2];
+        $this->dependencyPluginNames = preg_split("/, ?/", $matches[3]);
```
Comment (matthi4s) [inline src/Analysis/Problem/Bukkit/PluginDependenciesProblem.php:94]: Please add a test case for multiple dependencies, even if it's made up for now.

**Review by matthi4s — CHANGES_REQUESTED** [review]

File: src/Analysis/Problem/Bukkit/PluginDependenciesProblem.php [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+
+    /**
+     * @param static $insight
+     * @return bool
+     */
+    public function isEqual($insight): bool
+    {
+        return $this->getPluginName() === $insight->getPluginName()
+            && $this->getPluginPath() === $insight->getPluginPath()
+            && $this->dependencyPluginNames == $insight->dependencyPluginNames;
```
Comment (matthi4s) [inline src/Analysis/Problem/Bukkit/PluginDependenciesProblem.php:110]: This tries to use a protected property and also doesn't compare the array independent of the order of the plugin names.

**Review by matthi4s — CHANGES_REQUESTED** [review]

**Comment by JulianVennen** [comment]
I don't understand why the one of the CI jobs is failing while it's working on the other CI job and on my machine.
Since it's only complaining about some time issues in bedrock, it probably has nothing to do with the PR.
I'm guessing that some of GitHub's test runners run in a different time zone?

File: src/Analysis/Problem/Bukkit/PluginDependenciesProblem.php [hunk 4]
```diff
@@ -105,8 +114,20 @@ public function setMatches(array $matches, $patternKey): void
...
+        foreach ($this->getDependencyPlugins() as $plugin) {
+            if (!in_array($plugin, $dependencies)) {
+                return false;
+            }
+            $dependencies = array_filter($dependencies, function ($value) use ($plugin) {
+                return $value !== $plugin;
+            });
+        }
+
+        return sizeof($dependencies) === 0;
```
Comment (matthi4s) [inline src/Analysis/Problem/Bukkit/PluginDependenciesProblem.php:131]:
```suggestion
        return count($dependencies) === 0;
```

File: src/Analysis/Problem/Bukkit/PluginDependenciesProblem.php [hunk 5]
```diff
@@ -105,8 +114,20 @@ public function setMatches(array $matches, $patternKey): void
...
     public function isEqual($insight): bool
     {
-        return $this->getPluginName() === $insight->getPluginName()
-            && $this->getPluginPath() === $insight->getPluginPath()
-            && $this->dependencyPluginNames == $insight->dependencyPluginNames;
+        if ($this->getPluginName() !== $insight->getPluginName() || $this->getPluginPath() !== $insight->getPluginPath()) {
+            return false;
+        }
+
+        $dependencies = $insight->getDependencyPluginNames();
```
Comment (matthi4s) [inline src/Analysis/Problem/Bukkit/PluginDependenciesProblem.php:121]:
```suggestion
        $dependencies = $insight->getDependencyPlugins();
```

File: src/Analysis/Problem/Bukkit/PluginDependenciesProblem.php [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+        foreach ($this->getDependencyPlugins() as $plugin) {
+            if (!in_array($plugin, $dependencies)) {
+                return false;
+            }
+            $dependencies = array_filter($dependencies, function ($value) use ($plugin) {
+                return $value !== $plugin;
+            });
+        }
```
Comment (matthi4s) [inline src/Analysis/Problem/Bukkit/PluginDependenciesProblem.php:129]: Sorting and comparing the arrays with `===` is probably easier.

File: src/Analysis/Problem/Bukkit/PluginDependenciesProblem.php [hunk 7]
```diff
@@ -0,0 +1,133 @@
...
+            $this->addSolution((new PluginInstallSolution())->setPluginName($name));
+        }
+        $this->addSolution((new FileDeleteSolution())->setRelativePath($this->getPluginPath()));
+    }
+
+    /**
+     * @param static $insight
+     * @return bool
+     */
+    public function isEqual($insight): bool
```
Comment (matthi4s) [inline src/Analysis/Problem/Bukkit/PluginDependenciesProblem.php:115]: Please also add a test case for this with two similar messages in a log.

**Review by matthi4s — CHANGES_REQUESTED** [review]

**Comment by matthi4s** [comment]
> I don't understand why the one of the CI jobs is failing while it's working on the other CI job and on my machine. Since it's only complaining about some time issues in bedrock, it probably has nothing to do with the PR. I'm guessing that some of GitHub's test runners run in a different time zone?

Should be fixed in https://github.com/aternosorg/codex-minecraft/commit/cfa55269d4ae36278dae73db9cb45dffa3422e93

File: src/Analysis/Problem/Bukkit/PluginDependenciesProblem.php (hunk 7 above)
Comment (JulianVennen) [reply src/Analysis/Problem/Bukkit/PluginDependenciesProblem.php:115]: Do you mean a test case with two equal erros?

Comment (matthi4s) [reply src/Analysis/Problem/Bukkit/PluginDependenciesProblem.php:115]: Yes

File: src/Analysis/Problem/Bukkit/PluginDependenciesProblem.php [hunk 8]
```diff
@@ -0,0 +1,127 @@
...
+    }
+
+    /**
+     * Get a human readable message
+     *
+     * @return string
+     */
+    public function getMessage(): string
+    {
+        return Translator::getInstance()->getTranslation("plugin-dependencies-problem", [
```
Comment (matthi4s) [inline src/Analysis/Problem/Bukkit/PluginDependenciesProblem.php:75]: When only a single plugin is listed the language variable for the single dependency should be used instead.

**Review by matthi4s — CHANGES_REQUESTED** [review]

**Review by matthi4s — APPROVED** [review]

**Pull request merged**